src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexRunner.cs

```diff
@@ -491,6 +463,12 @@ protected bool IsBoundary(int index, int startpos, int endpos)
                    (index < endpos && RegexCharClass.IsBoundaryWordChar(runtext![index]));
         }
 
+        internal bool IsBoundary(ReadOnlySpan<char> inputSpan, int index)
+        {
+            return (index > runtextbeg && RegexCharClass.IsBoundaryWordChar(inputSpan[index - 1])) !=
```

Comment:
Is runtextbeg here correct?  Shouldn't it be 0?  In which case my similar comments from the source generated code apply as well.

Same question below in IsECMABoundary.

Also, if it's supposed to be 0 rather than runtextbeg and no tests are failing, it seems we're missing some tests.

Reply:
Yes it can be 0 too. I don't think we are missing any tests, instead what is happening is that now that we are slicing a span from runtextbeg to runtextend always, we are setting runtextbeg to 0 in InitializeForScan method:

https://github.com/dotnet/runtime/blob/4ca5883d2a72424afa22ce4d7ae21adf5cf4e3b1/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexRunner.cs#L166-L173